src/macroxs.F90

```diff
@@ -0,0 +1,243 @@
+module macroxs
+
+  use constants
+  use macroxs_header,  only: MacroXS_Base, MacroXS_Iso, MacroXS_Angle, &
+                             expand_harmonic
+  use material_header, only: Material
+  use math
+  use nuclide_header,  only: find_angle, MaterialMacroXS, NuclideMicroXS, &
+                             Nuclide_MG, NuclideMGContainer
+  use random_lcg,      only: prn
+  use scattdata_header
+  use search
+
+  implicit none
+
+contains
+
+!===============================================================================
+! UPDATE_XS stores the xs to work with
+!===============================================================================
+
+  subroutine calculate_mgxs(this, gin, uvw, xs)
+    class(MacroXS_Base),   intent(in)    :: this
+    integer,               intent(in)    :: gin         ! Incoming neutron group
+    real(8),               intent(in)    :: uvw(3)      ! Incoming neutron direction
+    type(MaterialMacroXS), intent(inout) :: xs
```

Comment:
Understand on the preference for matching calculate_xs, this is similar to the above though in that to do it that way pulls in more dependencies than I need and though I dont remember which situation it was exactly anymore, but I had a big arm wrestling match with circular dependencies during the development of this.